src/routes/authRoutes.js

```diff
@@ -0,0 +1,52 @@
+const bcrypt = require('bcrypt')
+const { createUser, loginUser } = require('../schemas/schemas')
+
+module.exports = async function authenticateUsers(fastify) {
+  // Creates user and hash password for db
+  fastify.post('/auth', { schema: createUser }, async (request, reply) => {
+    const { username, password } = request.body
+    if (!request.body) {
+      return { message: 'No username or password provided' }
+    }
+    if (!username) {
+      return { message: 'No username provided' }
+    }
+    bcrypt.hash(password, 10, async (err, hash) => {
+      if (err) {
+        return reply.code(500).send({ message: 'An error occured while hashing password.' })
+      }
+
+      const client = await fastify.pg.connect()
+      const { rows } = await client.query(
+        'INSERT INTO users (username, password) VALUES ($1, $2) RETURNING *;',
+        [username, hash]
+      )
+      client.release()
+      return rows
```

Comment:
Instead of just returning rows you should return a success message to indicate that the user was created successfully.